core/src/main/java/org/elasticsearch/index/query/ScriptQueryBuilder.java

```diff
@@ -31,12 +41,21 @@
 
     static final ScriptQueryBuilder PROTOTYPE = new ScriptQueryBuilder((Script) null);
 
-    private Script script;
+    private final Script script;
 
     public ScriptQueryBuilder(Script script) {
         this.script = script;
```

Comment:
given that the parser will never pass in null here, I think we could add a null check here as we did in other queries, so that if somebod using the java api sets null will get back error immediately. then we could remove the check from validate